### ISSUE TITLE
Boolean inputs with question mark syntax behave inappropriately

### ISSUE THREAD

**Issue by supremebeing7** [orgsync/active_interaction#367]
We have a service with some boolean filters, and passing them in works fine except that using question mark syntax gives the wrong value. Example:

    Templates::CleanHTML.run!(clean_single_smart: false)

In the service:

    boolean :clean_single_smart,  default: true

    def execute
      binding.pry
    end
    [2] pry(#<Templates::CleanHTML>)> clean_single_smart
    => false
    [3] pry(#<Templates::CleanHTML>)> clean_single_smart?
    => true

Removing the default has no effect either. 

Is this expected behavior? It seems to me it would be better if it either 1) worked the same with or without the question mark syntax, or 2) raised a `NoMethodError` like any other undefined booleans, such as:

    [1] pry(#<Templates::CleanHTML>)> test?
    NoMethodError: undefined method `test?' for #<Templates::CleanHTML:0x007fc5072f6bf0>

**Comment by AaronLasseigne** [comment]
That is the expected behavior. The predicate method indicates that an input has a value. https://github.com/orgsync/active_interaction#predicates

**Comment by tfausak** [comment]
This is expected behavior. [The documentation](https://github.com/orgsync/active_interaction/tree/v3.1.1#predicates) says:

> ActiveInteraction creates a predicate method for every input defined by a filter. So if you have an input called `foo`, there will be a predicate method called `#foo?`. That method will tell you if the input was given (that is, if it was not `nil`).

This can be confusing when used with boolean filters. 

- If you simply want the value of the input, use the plain attribute getter (`clean_single_smart`). 
- If you want to know if the value is not `nil`, use the predicate (`clean_single_smart?`). 
- If you want to know if the value was given in the input hash, use `given?` (`given?(:clean_single_smart)`). 

Here is a complete example:

``` rb
require 'active_interaction' # 3.1.1

class Example < ActiveInteraction::Base
  boolean :a
  boolean :b, default: nil
  boolean :c, default: false
  boolean :d, default: true

  def execute
    inputs.each do |name, value|
      puts "#{name}        = #{value.inspect}"
      puts "#{name}?       = #{public_send("#{name}?")}"
      puts "given? #{name} = #{given? name}"
      puts
    end
  end
end

Example.run!(a: true)
# a        = true
# a?       = true
# given? a = true
# 
# b        = nil
# b?       = false
# given? b = false
# 
# c        = false
# c?       = true
# given? c = false
# 
# d        = true
# d?       = true
# given? d = false
```

**Comment by supremebeing7** [comment]
Ah! I understand. Thanks. I thought I had read all the docs carefully after using this the past 6 months, but clearly not carefully enough. 

I agree that it is confusing, though. Can I suggest adding a short note about booleans specifically? The documentation is certainly clear enough as-is, but because there is so much documentation, I find myself just doing `cmd+F` to search the page for the word "boolean." Since the predicate method is mainly only confusing with booleans, just having a short note there for searching purposes could help any others that run across this issue.

Either way, thanks for your hard work, it's a great gem!

**Comment by tfausak** [comment]
This was fixed by #368.
